autoload/wiki/link.vim

```diff
@@ -242,6 +242,20 @@ function! wiki#link#template_md(url, ...) abort " {{{1
   return '[' . l:text . '](' . a:url . ')'
 endfunction
 
+" }}}1
+function! wiki#link#pick_template_type(url, ...) abort
+  "
+  " Pick the relevant link template command to use based on the users
+  " settings. Default to the wiki style one if its not set.
+  "
+  if !empty(g:wiki_link_target_type) &&
+    \ exists('*wiki#link#template_' . g:wiki_link_target_type)
+    return call('wiki#link#template_' . g:wiki_link_target_type, [a:url] + a:000)
+  else
+    return call ('wiki#link#template_wiki', [a:url] + a:000)
+  endif
+endfunction
```

Comment:
I would use `template_pick_from_option` as the name and I would implement it more like this:

```vim
function! wiki#link#template_pick_from_option(...) abort
  let l:type = get(g:, 'wiki_link_target_type', 'wiki')

  try
    return call('wiki#link#template_' . l:type, a:000)
  catch /E117/
    echoerr 'Link target type does not exist:' l:type
    echoerr 'See ":help g:wiki_link_target_type" for help.'
  endtry
endfunction
```

Reply:
Although: The option should be initialized when `wiki.vim` is initialized, so it seems to be not necessary to use the `get()` function.